commit a02ff7df6a3b4caf7cf9dfb7814a70c62642af9b
Author: Bjørn Mork <bjorn@mork.no>
Date:   Mon Sep 17 22:00:24 2018 +0200

    qmi_wwan: set DTR for modems in forced USB2 mode
    
    [ Upstream commit 922005c7f50e7f4b2a6dbc182e9c575b4f92396b ]
    
    Recent firmware revisions have added the ability to force
    these modems to USB2 mode, hiding their SuperSpeed
    capabilities from the host.  The driver has been using the
    SuperSpeed capability, as shown by the bcdUSB field of the
    device descriptor, to detect the need to enable the DTR
    quirk.  This method fails when the modems are forced to
    USB2 mode by the modem firmware.
    
    Fix by unconditionally enabling the DTR quirk for the
    affected device IDs.
    
    Reported-by: Fred Veldini <fred.veldini@gmail.com>
    Reported-by: Deshu Wen <dwen@sierrawireless.com>
    Signed-off-by: Bjørn Mork <bjorn@mork.no>
    Reported-by: Fred Veldini <fred.veldini@gmail.com>
    Reported-by: Deshu Wen <dwen@sierrawireless.com>
    Signed-off-by: Bjørn Mork <bjorn@mork.no>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/usb/qmi_wwan.c b/drivers/net/usb/qmi_wwan.c
index cb0cc30c3d6a..1e95d37c6e27 100644
--- a/drivers/net/usb/qmi_wwan.c
+++ b/drivers/net/usb/qmi_wwan.c
@@ -1206,13 +1206,13 @@ static const struct usb_device_id products[] = {
 	{QMI_FIXED_INTF(0x1199, 0x9061, 8)},	/* Sierra Wireless Modem */
 	{QMI_FIXED_INTF(0x1199, 0x9063, 8)},	/* Sierra Wireless EM7305 */
 	{QMI_FIXED_INTF(0x1199, 0x9063, 10)},	/* Sierra Wireless EM7305 */
-	{QMI_FIXED_INTF(0x1199, 0x9071, 8)},	/* Sierra Wireless MC74xx */
-	{QMI_FIXED_INTF(0x1199, 0x9071, 10)},	/* Sierra Wireless MC74xx */
-	{QMI_FIXED_INTF(0x1199, 0x9079, 8)},	/* Sierra Wireless EM74xx */
-	{QMI_FIXED_INTF(0x1199, 0x9079, 10)},	/* Sierra Wireless EM74xx */
-	{QMI_FIXED_INTF(0x1199, 0x907b, 8)},	/* Sierra Wireless EM74xx */
-	{QMI_FIXED_INTF(0x1199, 0x907b, 10)},	/* Sierra Wireless EM74xx */
-	{QMI_FIXED_INTF(0x1199, 0x9091, 8)},	/* Sierra Wireless EM7565 */
+	{QMI_QUIRK_SET_DTR(0x1199, 0x9071, 8)},	/* Sierra Wireless MC74xx */
+	{QMI_QUIRK_SET_DTR(0x1199, 0x9071, 10)},/* Sierra Wireless MC74xx */
+	{QMI_QUIRK_SET_DTR(0x1199, 0x9079, 8)},	/* Sierra Wireless EM74xx */
+	{QMI_QUIRK_SET_DTR(0x1199, 0x9079, 10)},/* Sierra Wireless EM74xx */
+	{QMI_QUIRK_SET_DTR(0x1199, 0x907b, 8)},	/* Sierra Wireless EM74xx */
+	{QMI_QUIRK_SET_DTR(0x1199, 0x907b, 10)},/* Sierra Wireless EM74xx */
+	{QMI_QUIRK_SET_DTR(0x1199, 0x9091, 8)},	/* Sierra Wireless EM7565 */
 	{QMI_FIXED_INTF(0x1bbb, 0x011e, 4)},	/* Telekom Speedstick LTE II (Alcatel One Touch L100V LTE) */
 	{QMI_FIXED_INTF(0x1bbb, 0x0203, 2)},	/* Alcatel L800MA */
 	{QMI_FIXED_INTF(0x2357, 0x0201, 4)},	/* TP-LINK HSUPA Modem MA180 */
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1005_a02ff7df6a3b4caf7cf9dfb7814a70c62642af9b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1005_a02ff7df6a3b4caf7cf9dfb7814a70c62642af9b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
